crates/hir/src/lib.rs

```diff
@@ -2220,13 +2219,17 @@ impl LifetimeParam {
 
 #[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
 pub struct ConstParam {
-    pub(crate) id: ConstParamId,
+    pub(crate) id: TypeOrConstParamId,
 }
 
 impl ConstParam {
+    pub fn merge(self) -> TypeOrConstParam {
```

Comment:
Feels like this should be just a plain `From` impl?

Reply:
Though I guess the benefit here is in not having a generic output type involved, the name `merge` feels wrong though since we don't merge anything

Reply:
`merge` is reverse of `split` on the `TypeOrConstParam`. Suggestion for better name? `generalize`?

Reply:
I don't have better names either, so keeping it as merge or generalize is fine

Reply:
I would also expect this to just be a `From` impl, is that not possible?